src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -85,8 +93,11 @@ public DataFetcherResult getRecords(int maxRecords) {
     final DataFetcherResult TERMINAL_RESULT = new DataFetcherResult() {
         @Override
         public GetRecordsResult getResult() {
-            return new GetRecordsResult().withMillisBehindLatest(null).withRecords(Collections.emptyList())
-                    .withNextShardIterator(null);
+            return new GetRecordsResult()
+                    .withMillisBehindLatest(null)
+                    .withRecords(Collections.emptyList())
+                    .withNextShardIterator(null)
+                    .withChildShards(Collections.emptyList());
```

Comment:
Would we need to recover the actual child shards somehow? For example if we hit the TERMINAL_RESULT in line 89, how will we discover the child shards? Or will we have already collected them in a prior ADVANCING_RESULT?